### PR TITLE
update assemblyVersion, packageIndex & resources data

### PR DESCRIPTION
**Pull request by Anipik**
Working Towards https://github.com/dotnet/runtime/issues/44078

- Update AssemblyVersion to 6.0.0.0
- Update the packageIndex 
- Update the Resources TestData Extensions

### CODE REVIEW

**Review by ViktorHofer — COMMENTED** [review]
@ericstj @joperezr pls review the packageIndex changes.

File: src/libraries/pkg/baseline/packageIndex.json [hunk 1]
```diff
@@ -28,7 +35,7 @@
       "InboxOn": {},
       "AssemblyVersionInPackageVersion": {
         "1.0.0.0": "1.0.0",
-        "5.0.0.0": "6.0.0"
+        "6.0.0.0": "6.0.0"
```
Comment (ericstj) [inline src/libraries/pkg/baseline/packageIndex.json:38]: This and many below are missing entries for "5.0.0.0": "5.0.0".   I think everything that existed during 5.0 timeframe should get both entries.

File: src/libraries/pkg/baseline/packageIndex.json [hunk 2]
```diff
@@ -10,6 +10,13 @@
         "net45": "4.0.0.0"
       }
     },
+    "dotnet-ilverify": {
```
Comment (ericstj) [inline src/libraries/pkg/baseline/packageIndex.json:13]: I wouldn't expect any *additions* to the index as a result of this work.  You aren't adding new packages to the repo.  You should undo any new entries your process is creating.

**Review by ericstj — CHANGES_REQUESTED** [review]

File: src/libraries/pkg/baseline/packageIndex.json (hunk 1 above)
Comment (joperezr) [reply src/libraries/pkg/baseline/packageIndex.json:38]: And looks like most of these are also missing 5.0.0 to be listed as Stable Version. Is that going to happen after we ship or should that be includded on this PR as well?

File: src/libraries/pkg/baseline/packageIndex.json [hunk 3]
```diff
@@ -1463,6 +1488,15 @@
       "BaselineVersion": "6.0.0",
       "InboxOn": {}
     },
+    "runtime.win-x64.Microsoft.NETCore.ILAsm": {
```
Comment (joperezr) [inline src/libraries/pkg/baseline/packageIndex.json:1491]: I suppose the way you generated these changes was by pointing to a local folder of your build? If so I guess that would explain the additions but as Eric said we should probably remove those entries.

File: src/libraries/pkg/baseline/packageIndex.json (hunk 1 above)
Comment (ericstj) [reply src/libraries/pkg/baseline/packageIndex.json:38]: That should be included here.

**Comment by joperezr** [comment]
What changed in the testdata resources as that can't be viewed here? and why did that have to change?

File: src/libraries/pkg/baseline/packageIndex.json (hunk 1 above)
Comment (Anipik) [reply src/libraries/pkg/baseline/packageIndex.json:38]: Yes i am planning to include that after we release tomorrow

File: eng/Versions.props [hunk 4]
```diff
@@ -12,7 +12,7 @@
          as for the patches and revisions should be manually updated per assembly if it is serviced. -->
     <!-- Changing assembly versions without changing TFM lead to broken packages.
          This shoud be incremented after introducing the 6.0 TFM. -->
-    <AssemblyVersion>5.0.0.0</AssemblyVersion>
+    <AssemblyVersion>6.0.0.0</AssemblyVersion>
```
Comment (safern) [inline eng/Versions.props:15]: I remember there were some assemblies that overrode this that couldn't be versioned as 5.0.0.0, right? Do we need to bump those?

File: src/libraries/System.Formats.Cbor/ref/System.Formats.Cbor.csproj [hunk 5]
```diff
@@ -1,13 +1,13 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>$(NetCoreAppCurrent)</TargetFrameworks>
+    <TargetFrameworks>net5.0</TargetFrameworks>
```
Comment (ViktorHofer) [inline src/libraries/System.Formats.Cbor/ref/System.Formats.Cbor.csproj:3]: why?

Comment (ViktorHofer) [reply src/libraries/System.Formats.Cbor/ref/System.Formats.Cbor.csproj:3]: Just saw we shipped the package with net5 yesterday. Was asking as I thought that you handled these configuration changes when you updated the tfm to net6.

Comment (safern) [reply src/libraries/System.Formats.Cbor/ref/System.Formats.Cbor.csproj:3]: Yeah. it seems like it slipped through as the package index wasn't complete and hence it didn't fail because this package was knew and there were not inferred supported TFMs from previous packages, and now the build was failing saying that there was no compile assets for 5.0 for this package which was supported by the previous package.

Comment (ViktorHofer) [reply src/libraries/System.Formats.Cbor/ref/System.Formats.Cbor.csproj:3]: That makes sense, thanks for explaining.

File: src/libraries/pkg/baseline/packageIndex.json [hunk 6]
```diff
@@ -4208,10 +4208,10 @@
     },
     "System.Net.Quic": {
       "InboxOn": {
+        "netcoreapp3.0": "4.2.1.0",
```
Comment (safern) [inline src/libraries/pkg/baseline/packageIndex.json:4211]: Why was this added? I think `System.Net.Quic` is new and will be inbox starting in `net6.0`

File: src/libraries/pkg/baseline/packageIndex.json [hunk 7]
```diff
@@ -4208,10 +4208,10 @@
...
         "net5.0": "5.0.0.0"
```
Comment (safern) [inline src/libraries/pkg/baseline/packageIndex.json:4212]: I also don't see it on `net5.0`: https://github.com/dotnet/runtime/blob/release/5.0/src/libraries/NetCoreAppLibrary.props

**Comment by Anipik** [comment]
>  Restored C:\h\w\A8170921\w\A09A08ED\u\net472\project.csproj (in 16.28 sec).
  Determining projects to restore...
  Restored C:\h\w\A8170921\w\A09A08ED\u\net5.0\project.csproj (in 48.6 sec).
  Determining projects to restore...
C:\h\w\A8170921\w\A09A08ED\u\net6.0\project.csproj : error NU1102: Unable to find package System.ComponentModel with version (>= 6.0.0-ci)
C:\h\w\A8170921\w\A09A08ED\u\net6.0\project.csproj : error NU1102:   - Found 14 version(s) in nuget.org [ Nearest version: 4.3.0 ]
C:\h\w\A8170921\w\A09A08ED\u\net6.0\project.csproj : error NU1102:   - Found 3 version(s) in https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json [ Nearest version: 4.3.0 ]
C:\h\w\A8170921\w\A09A08ED\u\net6.0\project.csproj : error NU1102:   - Found 0 version(s) in C:\h\w\A8170921\p\packages
C:\h\w\A8170921\w\A09A08ED\u\net6.0\project.csproj : error NU1102: Unable to find package System.Reflection.Emit.ILGeneration with version (>= 6.0.0-ci)
C:\h\w\A8170921\w\A09A08ED\u\net6.0\project.csproj : error NU1102:   - Found 2223 version(s) in https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json [ Nearest version: 5.0.0-alpha1.19523.8 ]
C:\h\w\A8170921\w\A09A08ED\u\net6.0\project.csproj : error NU1102:   - Found 31 version(s) in nuget.org [ Nearest version: 4.7.0 ]
C:\h\w\A8170921\w\A09A08ED\u\net6.0\project.csproj : error NU1102:   - Found 0 version(s) in C:\h\w\A8170921\p\packages
C:\h\w\A8170921\w\A09A08ED\u\net6.0\project.csproj : error NU1102: Unable to find package System.Reflection.Emit.Lightweight with version (>= 6.0.0-ci)
C:\h\w\A8170921\w\A09A08ED\u\net6.0\project.csproj : error NU1102:   - Found 2222 version(s) in https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json [ Nearest version: 5.0.0-alpha1.19523.8 ]
C:\h\w\A8170921\w\A09A08ED\u\net6.0\project.csproj : error NU1102:   - Found 31 version(s) in nuget.org [ Nearest version: 4.7.0 ]
C:\h\w\A8170921\w\A09A08ED\u\net6.0\project.csproj : error NU1102:   - Found 0 version(s) in C:\h\w\A8170921\p\packages
C:\h\w\A8170921\w\A09A08ED\u\net6.0\project.csproj : error NU1102: Unable to find package System.Threading.ThreadPool with version (>= 6.0.0-ci)
C:\h\w\A8170921\w\A09A08ED\u\net6.0\project.csproj : error NU1102:   - Found 14 version(s) in nuget.org [ Nearest version: 4.3.0 ]
C:\h\w\A8170921\w\A09A08ED\u\net6.0\project.csproj : error NU1102:   - Found 2 version(s) in https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json [ Nearest version: 4.3.0 ]
C:\h\w\A8170921\w\A09A08ED\u\net6.0\project.csproj : error NU1102:   - Found 0 version(s) in C:\h\w\A8170921\p\packages


@ericstj any idea what is causing this ?

File: src/libraries/pkg/baseline/packageIndex.json (hunk 6 above)
Comment (Anipik) [reply src/libraries/pkg/baseline/packageIndex.json:4211]: i use the packageindex from release/5.0 branch as base and it seems like we have system.net.quic there in packageindex but not in repo https://github.com/dotnet/runtime/blob/release/5.0/src/libraries/pkg/baseline/packageIndex.json#L4013

i will remove this.

File: eng/Versions.props (hunk 4 above)
Comment (Anipik) [reply eng/Versions.props:15]: They still need to be pinned to match the versions on desktop

File: src/libraries/System.IO.Pipes/ref/System.IO.Pipes.csproj [hunk 8]
```diff
@@ -1,12 +1,16 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>$(NetCoreAppCurrent)</TargetFrameworks>
+    <TargetFrameworks>$(NetCoreAppCurrent);net5.0;</TargetFrameworks>
```
Comment (ericstj) [inline src/libraries/System.IO.Pipes/ref/System.IO.Pipes.csproj:3]: You need to remove this. Pipes is part of the shared framework, not a package.  We shouldn't rebuild for old TFMs for things that aren't packages, since those builds would never ship.

File: src/libraries/System.IO.Pipes/src/System.IO.Pipes.csproj [hunk 9]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <OmitTransitiveCompileReferences>true</OmitTransitiveCompileReferences>
-    <TargetFrameworks>$(NetCoreAppCurrent)-windows;$(NetCoreAppCurrent)-Unix;$(NetCoreAppCurrent)</TargetFrameworks>
+    <TargetFrameworks>$(NetCoreAppCurrent)-windows;$(NetCoreAppCurrent)-Unix;$(NetCoreAppCurrent);net5.0;net5.0-windows;net5.0-Unix</TargetFrameworks>
```
Comment (ericstj) [inline src/libraries/System.IO.Pipes/src/System.IO.Pipes.csproj:5]: Remove these as well.

File: src/libraries/System.IO.Pipes.AccessControl/src/System.IO.Pipes.AccessControl.csproj [hunk 10]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <Nullable>enable</Nullable>
-    <TargetFrameworks>$(NetCoreAppCurrent)-windows;$(NetCoreAppCurrent)</TargetFrameworks>
+    <TargetFrameworks>$(NetCoreAppCurrent)-windows;$(NetCoreAppCurrent);net5.0;net5.0-windows</TargetFrameworks>
```
Comment (ericstj) [inline src/libraries/System.IO.Pipes.AccessControl/src/System.IO.Pipes.AccessControl.csproj:4]: Make sure that this doesn't use any ProjectReferences on `net5.0`.  It should build against the ref-pack in that case.

File: src/libraries/System.IO.Pipes/ref/System.IO.Pipes.csproj (hunk 8 above)
Comment (ericstj) [reply src/libraries/System.IO.Pipes/ref/System.IO.Pipes.csproj:3]: We could add an error to our build system to catch this problem.

File: eng/Versions.props (hunk 4 above)
Comment (ericstj) [reply eng/Versions.props:15]: I think @safern was asking about things which weren't able to use the product version.  Here's one: https://github.com/dotnet/runtime/blob/466e4b78e4a06a73ea6cbe24324c6256fc546f58/src/libraries/Microsoft.VisualBasic.Core/Directory.Build.props#L6

Comment (ericstj) [reply eng/Versions.props:15]: Maybe this one too:
https://github.com/dotnet/runtime/blob/466e4b78e4a06a73ea6cbe24324c6256fc546f58/src/libraries/System.Reflection.Context/Directory.Build.props#L4

File: src/libraries/System.Security.Principal.Windows/src/System.Security.Principal.Windows.csproj [hunk 11]
```diff
@@ -18,7 +18,7 @@
     <Nullable>annotations</Nullable>
   </PropertyGroup>
 
-  <ItemGroup Condition="($(TargetFramework.StartsWith('$(NetCoreAppCurrent)')) or $(TargetFramework.StartsWith('netcoreapp2'))) and '$(TargetsWindows)' == 'true'">
+  <ItemGroup Condition="($(TargetFramework.StartsWith('$(NetCoreAppCurrent)')) or $(TargetFramework.StartsWith('netcoreapp2.1')) or '$(TargetFramework)' == 'netstandard2.0') and '$(TargetsWindows)' == 'true'">
```
Comment (ericstj) [inline src/libraries/System.Security.Principal.Windows/src/System.Security.Principal.Windows.csproj:21]: nit: this could probably be simplified to just not net4x.

File: src/libraries/pkg/test/frameworkSettings/net6.0/settings.targets [hunk 12]
```diff
@@ -1,6 +1,8 @@
 <Project>
   <PropertyGroup>
     <RuntimeFrameworkVersion>$(MicrosoftNETCoreAppVersion)</RuntimeFrameworkVersion>
+    <!-- This should be enabled once we have updated assembly versions. -->
```
Comment (ericstj) [inline src/libraries/pkg/test/frameworkSettings/net6.0/settings.targets:4]: Make sure we have an issue so we don't forget.

**Review by ericstj — APPROVED** [review]

File: eng/Versions.props (hunk 4 above)
Comment (ViktorHofer) [reply eng/Versions.props:15]: What's the reason that System.Reflection.Context wasn't updated to 5.0.0?

Comment (ViktorHofer) [reply eng/Versions.props:15]: Hmm I assume it's the same reason as for System.ValueTuple?

>

Comment (ericstj) [reply eng/Versions.props:15]: If it's a 1:1 assembly with desktop and exposes a netstandard ref we need to make sure the reference assembly is pinned at 4.0.0.0 and implementation can still rev.

Comment (ViktorHofer) [reply eng/Versions.props:15]: Yes, the reference assembly is pinned to 4.0.0.0, so I'm wondering why the src wasn't bumped to 5.0.0.0.

File: src/libraries/pkg/test/frameworkSettings/net6.0/settings.targets (hunk 12 above)
Comment (Anipik) [reply src/libraries/pkg/test/frameworkSettings/net6.0/settings.targets:4]: Yeah we are tracking this here https://github.com/dotnet/runtime/issues/44078

File: eng/Versions.props (hunk 4 above)
Comment (Anipik) [reply eng/Versions.props:15]: yes, they need to have similar major and minor digits.

File: src/libraries/pkg/test/frameworkSettings/net6.0/settings.targets (hunk 12 above)
Comment (safern) [reply src/libraries/pkg/test/frameworkSettings/net6.0/settings.targets:4]: But this PR will close that issue 😄

File: src/libraries/System.Security.AccessControl/pkg/System.Security.AccessControl.pkgproj [hunk 13]
```diff
@@ -2,12 +2,8 @@
   <Import Project="$([MSBuild]::GetPathOfFileAbove(Directory.Build.props))" />
   <ItemGroup>
     <ProjectReference Include="..\ref\System.Security.AccessControl.csproj">
-      <SupportedFramework>net461;netcoreapp2.0;$(AllXamarinFrameworks)</SupportedFramework>
+      <SupportedFramework>net461;netcoreapp2.0;uap10.0.16299;$(AllXamarinFrameworks)</SupportedFramework>
     </ProjectReference>
-    <File Include="$(PlaceHolderFile)">
-      <TargetPath>runtimes/win/lib/uap10.0.16299</TargetPath>
-    </File>
-    <InboxOnTargetFramework Include="uap10.0.16299" />
```
Comment (ViktorHofer) [inline src/libraries/System.Security.AccessControl/pkg/System.Security.AccessControl.pkgproj:10]: why are removing the placeholder for uap?

File: src/libraries/System.Security.Principal.Windows/src/System.Security.Principal.Windows.csproj [hunk 14]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <TargetFrameworks>$(NetCoreAppCurrent)-windows;$(NetCoreAppCurrent)-Unix;netstandard2.0;netcoreapp2.0-windows;netcoreapp2.0-Unix;netcoreapp2.1-windows;netcoreapp2.1-Unix;net461-windows</TargetFrameworks>
+    <TargetFrameworks>$(NetCoreAppCurrent)-windows;$(NetCoreAppCurrent)-Unix;netstandard2.0;netstandard2.0-windows;netstandard2.0-Unix;netcoreapp2.1-windows;netcoreapp2.1-Unix;net461-windows</TargetFrameworks>
```
Comment (ViktorHofer) [inline src/libraries/System.Security.Principal.Windows/src/System.Security.Principal.Windows.csproj:4]: Why are we changing the tfm?

**Comment by Anipik** [comment]
sry viktor, your comment on the last commit got deleted with force push

File: src/libraries/System.Security.Principal.Windows/src/System.Security.Principal.Windows.csproj (hunk 14 above)
Comment (Anipik) [reply src/libraries/System.Security.Principal.Windows/src/System.Security.Principal.Windows.csproj:4]: The reason behind is to have netstandard2.0 implementation for uap.

Comment (ViktorHofer) [reply src/libraries/System.Security.Principal.Windows/src/System.Security.Principal.Windows.csproj:4]: but why are we doing this? uap isn't a configuration we are actively working on anymore.

File: src/libraries/Microsoft.VisualBasic.Core/Directory.Build.props [hunk 15]
```diff
@@ -1,9 +1,7 @@
 ﻿<Project>
   <Import Project="..\Directory.Build.props" />
   <PropertyGroup>
-    <MajorVersion>$([MSBuild]::Add($(MajorVersion), 6))</MajorVersion>
-    <PackageVersion>$(MajorVersion).$(MinorVersion).$(PatchVersion)</PackageVersion>
-    <AssemblyVersion>10.0.6.0</AssemblyVersion>
+    <AssemblyVersion>$([MSBuild]::Add($(MajorVersion), 6)).$(MinorVersion).0.0</AssemblyVersion>
```
Comment (ViktorHofer) [inline src/libraries/Microsoft.VisualBasic.Core/Directory.Build.props]: The previous AssemblyVersion was 10.x.x.x. This will now produce 12.x.x.x. What about 11.x.x.x? Intentionally left out? 😄

Comment (Anipik) [reply src/libraries/Microsoft.VisualBasic.Core/Directory.Build.props]: just changed a second before ur comment

File: src/libraries/System.Security.Principal.Windows/src/System.Security.Principal.Windows.csproj (hunk 14 above)
Comment (Anipik) [reply src/libraries/System.Security.Principal.Windows/src/System.Security.Principal.Windows.csproj:4]: we are still running package testing for it.

File: src/libraries/System.Security.AccessControl/pkg/System.Security.AccessControl.pkgproj (hunk 13 above)
Comment (Anipik) [reply src/libraries/System.Security.AccessControl/pkg/System.Security.AccessControl.pkgproj:10]: @ericstj might be able to better answer this one

File: src/libraries/Microsoft.VisualBasic.Core/Directory.Build.props (hunk 15 above)
Comment (ViktorHofer) [reply src/libraries/Microsoft.VisualBasic.Core/Directory.Build.props]: Wondering if we should just increment the patch version instead so that we use the repo wide 11.x.x.x version in 6 years (lol).

Comment (ViktorHofer) [reply src/libraries/Microsoft.VisualBasic.Core/Directory.Build.props]: I don't think this is important at all^^

File: src/libraries/System.Security.Principal.Windows/src/System.Security.Principal.Windows.csproj (hunk 14 above)
Comment (ViktorHofer) [reply src/libraries/System.Security.Principal.Windows/src/System.Security.Principal.Windows.csproj:4]: IMO we should remove package testing in master for it and/or remove the uap assets in our packages. (tomorrow's meeting)

Comment (Anipik) [reply src/libraries/System.Security.Principal.Windows/src/System.Security.Principal.Windows.csproj:4]: cc @ericstj

File: src/libraries/System.Security.AccessControl/pkg/System.Security.AccessControl.pkgproj (hunk 13 above)
Comment (ericstj) [reply src/libraries/System.Security.AccessControl/pkg/System.Security.AccessControl.pkgproj:10]: These packages were actually broken for UAP in last release (and maybe even previous) because the netstandard ref was versioned while UAP still got a placeholder (and thus would provide the older version inbox assembly).  The use of place holders is a relic of the old pre-conflict-resolution way of dealing with assemblies that were part of the framework.  We can remove that placeholder and simplify these cases while fixing the netstandard consumption case.  Worst case that happens here is that the implementation provided by the package is somehow worse than an old UAP configuration that was in the meta-package itself.  That's actually no worse than the type of breaks we already made when removing all the UAP package configurations from the repo.

**Comment by ericstj** [comment]
Woo hoo! Thank you @Anipik

**Comment by Anipik** [comment]
> Woo hoo! Thank you @Anipik

Finally everything is in net6 mode.
